src/overlays/actors/ovl_Bg_Hidan_Sekizou/z_bg_hidan_sekizou.c

```diff
@@ -110,25 +112,323 @@ static ColliderJntSphInit D_8088E258 = {
+                           gFireTempleFireball6Tex, gFireTempleFireball7Tex };
+
+void func_8088CEC0(BgHidanSekizou* this, s32 arg1, s16 arg2) {
+    s32 i;
+    s32 start = arg1 * 3;
+    s32 sp34 = start + 3;
+    f32 sp30 = Math_SinS(arg2);
+    f32 sp2C = Math_CosS(arg2);
+
+    for (i = start; i < sp34; i++) {
+        ColliderJntSphElement* element = &this->collider.elements[i];
+
+        element->dim.worldSphere.center.x = this->dyna.actor.home.pos.x + (sp2C * element->dim.modelSphere.center.x) +
+                                            (sp30 * element->dim.modelSphere.center.z);
+        element->dim.worldSphere.center.y = (s16)this->dyna.actor.home.pos.y + element->dim.modelSphere.center.y;
+        element->dim.worldSphere.center.z = this->dyna.actor.home.pos.z - (sp30 * element->dim.modelSphere.center.x) +
+                                            (sp2C * element->dim.modelSphere.center.z);
+        element->info.toucherFlags |= TOUCH_ON;
+        element->info.ocElemFlags |= OCELEM_ON;
+    }
+}
+
+void BgHidanSekizou_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    BgHidanSekizou* this = THIS;
+    s32 i;
+    CollisionHeader* sp40 = NULL;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DPM_UNK);
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->dyna.actor, &D_8088E258, this->elements);
+    for (i = 0; i < 6; i++) {
+        this->collider.elements[i].dim.worldSphere.radius = this->collider.elements[i].dim.modelSphere.radius;
+    }
+    if (this->dyna.actor.params == 0) {
+        this->unk_168[0] = 36;
+        for (i = 0; i < 2; i++) {
+            func_8088CEC0(this, i, this->dyna.actor.shape.rot.y + ((i == 0) ? 0x2000 : -0x2000));
+        }
+        CollisionHeader_GetVirtual(&gFireTempleStationaryFlamethrowerShortCol, &sp40);
+        this->updateFunc = func_8088D720;
+    } else {
+        this->unk_168[0] = this->unk_168[1] = this->unk_168[2] = this->unk_168[3] = 0;
+        CollisionHeader_GetVirtual(&gFireTempleStationaryFlamethrowerTallCol, &sp40);
+        this->updateFunc = func_8088D434;
+    }
+    this->unk_170 = 0;
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, sp40);
+    CollisionCheck_SetInfo(&this->dyna.actor.colChkInfo, NULL, &D_8088E268);
+}
+
+void BgHidanSekizou_Destroy(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    BgHidanSekizou* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyJntSph(globalCtx, &this->collider);
+}
+
+void func_8088D434(BgHidanSekizou* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    s32 i;
+    s32 isAligned[2];
+    s32 isClose;
+    s32 phi_s4;
+
+    isClose = (this->dyna.actor.xzDistToPlayer < 300.0f);
```

Comment:
```suggestion
    isClose = this->dyna.actor.xzDistToPlayer < 300.0f;
```
also this can probably be moved up